presto-spi/src/main/java/io/prestosql/spi/type/SqlTime.java

```diff
@@ -15,100 +15,80 @@
 
 import com.fasterxml.jackson.annotation.JsonValue;
 
-import java.time.Instant;
-import java.time.ZoneId;
-import java.time.ZoneOffset;
-import java.time.format.DateTimeFormatter;
 import java.util.Objects;
-import java.util.Optional;
+
+import static io.prestosql.spi.type.TimeType.MAX_PRECISION;
+import static io.prestosql.spi.type.Timestamps.MINUTES_PER_HOUR;
+import static io.prestosql.spi.type.Timestamps.PICOSECONDS_PER_HOUR;
+import static io.prestosql.spi.type.Timestamps.PICOSECONDS_PER_MINUTE;
+import static io.prestosql.spi.type.Timestamps.PICOSECONDS_PER_SECOND;
+import static io.prestosql.spi.type.Timestamps.POWERS_OF_TEN;
+import static io.prestosql.spi.type.Timestamps.SECONDS_PER_MINUTE;
+import static io.prestosql.spi.type.Timestamps.round;
+import static java.lang.String.format;
 
 public final class SqlTime
 {
-    private static final DateTimeFormatter JSON_FORMATTER = DateTimeFormatter.ofPattern("HH:mm:ss.SSS");
-
-    private final long millis;
-    private final Optional<TimeZoneKey> sessionTimeZoneKey;
-
-    public SqlTime(long millis)
-    {
-        this.millis = millis;
-        this.sessionTimeZoneKey = Optional.empty();
-    }
+    private final int precision;
+    private final long picos;
 
-    @Deprecated
-    public SqlTime(long millisUtc, TimeZoneKey sessionTimeZoneKey)
+    public static SqlTime newInstance(int precision, long picos)
     {
-        this.millis = millisUtc;
-        this.sessionTimeZoneKey = Optional.of(sessionTimeZoneKey);
+        return new SqlTime(precision, round(picos, 12 - precision));
```

Comment:
this  should not need to round, it should just "represent"

so maybe `checkArgument` instead?